File: back/node/app/core/data/repository/GameRepository.ts
import {PlayerType} from "../../model/PlayerType.js";
import {BoardType} from "../../model/BoardType.js";
import {GameType} from "../../model/GameType.js";
import { Result } from "../../common/result/Result.js";
import {CellPosition} from "../../model/CellPosition.js";
import {StateType} from "../../model/StateType.js";
import {GameInfo} from "../../database/redis/entity/GameInfo.js";

interface GameRepository {
    createNewGame(newKey: string, board: BoardType, player: PlayerType): Promise<Result<GameType>>;
    resetGame(gameID: string, board: BoardType, player: PlayerType): Promise<Result<GameType>>;
    updateBoard(gameID: string, position: CellPosition, player: PlayerType): Promise<Result<StateType>>;
    getCurrentPlayer(gameID: string): Promise<Result<PlayerType>>;
    getBoardState(gameID: string): Promise<Result<StateType>>;
    getGameState(gameID: string): Promise<Result<GameType>>;
    updateGameState(gameID: string, board: StateType, gameInfo: GameInfo): Promise<Result<GameType>>;
}

export {GameRepository}

File: back/kotlin/app/core/data/repository/GameRepository.kt
package core.data.repository

import core.common.result.Result
import core.model.StateType
import core.model.PlayerType
import core.model.CellPosition
import core.model.GameType
import core.database.redis.entity.GameInfo

interface GameRepository {
    suspend fun createNewGame(newKey: String, board: StateType, player: PlayerType): Result<GameType>
    suspend fun resetGame(gameID: String, board: StateType, player: PlayerType): Result<GameType>
    suspend fun updateBoard(gameID: String, position: CellPosition, player: PlayerType): Result<StateType>
    suspend fun getCurrentPlayer(gameID: String): Result<PlayerType>
    suspend fun getBoardState(gameID: String): Result<StateType>
    suspend fun getGameState(gameID: String): Result<GameType>
    suspend fun updateGameState(gameID: String, board: StateType, info: GameInfo): Result<GameType>
}